src/Lecture1.hs

```diff
@@ -123,4 +138,15 @@ and lower than 6 elements (4, 5, 6, 7, 8 and 9).
 
 🕯 HINT: Use recursion to implement this function.
 -}
-lowerAndGreater n list = error "TODO"
+lowerAndGreater :: Int -> [Int] -> String
+lowerAndGreater n list =
+    let lowerCount = countMatching (< n) list
+        greaterCount = countMatching (> n) list
```

Comment:
That's correct! 💯 

However, with this solution, you will walk through the list twice.
You can write it more performant with only one traversal of the list 🙂
It's desirable to write functions that traverse lists only a single time and consume elements only once. In that case, the composition of such functions will have nice streaming and lazy properties.

Reply:
True, thanks! Actually I was thinking on getting the counts with a foldl, but I'm really trying to do it recursively.

IIUC the new implementation will only traverse the list once. WDYT?